### PR TITLE
FEATURE: Make Neos independent from Setup package

### PR DESCRIPTION
**Pull request by bwaidelich**
Removes the `neos/setup` composer dependency from the `neos/neos` package
and adds a root dependency to the new `neos/neos-setup` package that now
contains configuration and custom "setup step" implementations.

Resolves: #2618

### CODE REVIEW

**Comment by bwaidelich** [comment]
*NOTE:* This requires https://github.com/neos/neos-setup to be registered with packagist.org

**Comment by daniellienert** [comment]
@bwaidelich: Is there a reason why you didn't registered it at packagist?

**Comment by daniellienert** [comment]
Seems it exists in the meantime.

File: composer.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
         "doctrine/dbal": "^2.8",
         "neos/party": "*",
         "neos/setup": "*",
+        "neos/neos-setup": "*",
```
Comment (skurfuerst) [inline composer.json]: hey,

IMHO this file is autogenerated; why did you add neos-setup here?

all the best,
Sebastian

**Comment by bwaidelich** [comment]
> IMHO composer.json at top level is autogenerated; why did you add neos-setup here?

@skurfuerst I added the new `neos/neos-setup` adapter package as dependency or how/when would that be auto-generated?
The `neos/setup` dependency could be removed unless we want to keep transitive dependencies (https://discuss.neos.io/t/composer-dependencies-declare-transitive-dependencies/1953)

**Comment by skurfuerst** [comment]
Hey @bwaidelich ,

As far as I understand the process for these monorepos is as follows:

- we hand-wrote the `.composer.json` files (note the dot prefix)
- we run https://github.com/neos/BuildEssentials/blob/master/ComposerManifestMerger.php - this takes the `.composer.json` and the composer.json of all packages, and emits a final `composer.json`.
- for every change in either the `composer.json` of a package, or the `.composer.json`, we have to re-run the tool.

In your case, you directly modified the generated composer.json without the input files (I think).

All the best,
Sebastian

**Comment by bwaidelich** [comment]
@skurfuerst Thanks for clarifying. I adjusted the composer template (so that `neos/neos-setup` is just a toplevel dependency that can be easily removed) and re-generated the `composer.json` file with provided script

**Comment by Benjamin-K** [comment]
Hey, is there any documentation about this / which packages can be removed after successful setup? I think this would help many people out there. Could be added to the installation setup docs: https://docs.neos.io/cms/installation-development-setup
